### PR TITLE
[Keyboard] Add Mechlovin9 rev.3

### PR DESCRIPTION
**Pull request by mechlovin**
## Description

Add Mechlovin9 Rev.3 PCB, APM32F103.
Thanks.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mechlovin/mechlovin9/rev3/info.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "keyboard_name": "Mechlovin9 Rev3",
+    "processor": "STM32F103",
+    "bootloader": "stm32duino",
+    "usb": {
+        "pid": "0x6509",
+        "device_version": "0.0.3"
+    }
```
Comment (waffle87) [inline keyboards/mechlovin/mechlovin9/rev3/info.json:8]:
```suggestion
    },
    "features": {
        "backlight": false
    },
    "bootmagic": {
        "matrix": [0, 13]
    },
    "matrix_pins": {
        "rows": ["B12", "B13", "B14", "B15", "A1"],
        "cols": ["B11", "B10", "B2", "B1", "B0", "A6", "A5", "A4", "A3", "C13", "B7, "B6", "B5", "B4", "B3"]
    },
    "indicators": {
        "caps_lock": "B9"
    }
```

File: keyboards/mechlovin/mechlovin9/rev3/config.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ */
+#define MATRIX_ROW_PINS { B12, B13, B14, B15, A1 }
+#define MATRIX_COL_PINS { B11, B10, B2, B1, B0, A6, A5, A4, A3, C13, B7, B6, B5, B4, B3 }
+
+#define LED_CAPS_LOCK_PIN B9
+
+/* Bootmagic Lite key configuration */
+ #define BOOTMAGIC_LITE_ROW 0
+ #define BOOTMAGIC_LITE_COLUMN 13
```
Comment (waffle87) [inline keyboards/mechlovin/mechlovin9/rev3/config.h:37]: This file should be removed with the above suggestion.

File: keyboards/mechlovin/mechlovin9/rev3/readme.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+* Keyboard Maintainer: [Team Mechlovin](https://github.com/mechlovin)
+* Hardware Supported: Kyuu Keyboard.
+* Hardware Availability: [Reddit GB](https://www.reddit.com/r/mechmarket/comments/i61par/gb_mechlovin_9_a_kyuu_65_keyboard_compatible_pcb/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make mechlovin/mechlovin9/rev3:default
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
```
Comment (waffle87) [inline keyboards/mechlovin/mechlovin9/rev3/readme.md:16]:
```suggestion
![mechlovin9](https://i.imgur.com/T5Lm6OYlh.png)

A 65% PCB, compatible with Kyuu.

* Keyboard Maintainer: [Team Mechlovin](https://github.com/mechlovin)
* Hardware Supported: Kyuu Keyboard.
* Hardware Availability: [Reddit GB](https://www.reddit.com/r/mechmarket/comments/i61par/gb_mechlovin_9_a_kyuu_65_keyboard_compatible_pcb/)

Make example for this keyboard (after setting up your build environment):

    make mechlovin/mechlovin9/rev3:default

Flashing example for this keyboard:

    make mechlovin/mechlovin9/rev3:default:flash

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

## Bootloader

```

File: keyboards/mechlovin/mechlovin9/rev3/rules.mk [hunk 4]
```diff
@@ -0,0 +1 @@
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
```
Comment (waffle87) [inline keyboards/mechlovin/mechlovin9/rev3/rules.mk:1]:
```suggestion
# This file intentionally left blank
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by drashna** [comment]
It looks like you have changes to the submodules here.  These need to be reverted before the PR can be approved. 

Rebasing sometimes will fix this. Otherwise, you'd want to do something like this to fix them:
```sh
git checkout master
git submodule foreach 'git log -n1 --oneline'
# remember the first abcdef1234 on the submodules that have an issue
git checkout @@your-branch@@
# for each of the problematic repo's:
cd lib/@@name@@
git checkout abcdef1234
```

**Review by drashna — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**